Abnormal Security Bug Fixes (#39104)
demisto/content: Packs/AbnormalSecurity/ReleaseNotes/2_3_0.md

#### Incident Fields

##### New: Abnormal Security Case GenAI Summary

- New: Added a new incident field- Abnormal Security Case GenAI Summary.

#### Integrations

##### Abnormal Security
- Updated the Docker image to: *demisto/python3:3.12.8.1983910*.

> - Added support for **Maximum incidents pages to fetch** parameter that maximum number of pages to fetch for incidents.
> - Added support for **Polling Lag Time (in minutes)** parameter that time in minutes to subtract from polling time window for data consistency.
> - Added support for *page_number* argument in the **abnormal-security-get-threat** command.
> - Added support for *page_size* argument in the **abnormal-security-get-threat** command.

Reviewer: ```suggestion
- Added support for the following parameters:
    - **Maximum incidents pages to fetch**
    - **Polling Lag Time (in minutes)**
- Added support for the following arguments in **abnormal-security-get-threat** command:
    - *page_number*
    - *page_size*
```

> 
> #### Incident Fields
> 
> ##### New: Abnormal Security Case GenAI Summary
> 
> - New: Added a new incident field- Abnormal Security Case GenAI Summary.
> 
> 
> #### Integrations
> 
> ##### Abnormal Security
> - Updated the Docker image to: *demisto/python3:3.12.8.1983910*.
> 
> - Newly added functionalities
>   - Added support for the following parameters:
>     - **Maximum incidents pages to fetch**
>       - Allows user to limit the maximum incidents to fetch in single call to fetch_incidents
>     - **Polling Lag Time (in minutes)**
>       - Allows user to add a custom lag while polling incidents
>   - Added support for the following arguments in **abnormal-security-get-threat** command:
>     - *page_number*
>     - *page_size*
> - Bug Fixes
>   - Fixed timestamp parsing to handle high-precision timestamps (nanoseconds) in incident data, ensuring proper handling of remediation and event timestamps.
> - Performance Optimization(s)
>   - Fixed pagination issue in fetch-incidents command to properly handle fetching of incidents when max_fetch is less than 100, optimizing API calls for better performance
>   - Enhanced pagination support in fetch-incidents to reliably fetch more than 100 incidents across multiple pages, enabling comprehensive data retrieval for large-scale incident analysis.
>   - Fixed pagination issue in get-threats command to properly handle fetching of multiple messages part of the threat by paginating and fetching all pages, while filtering the messages by user provided time window.
> 
>   
> #### Mappers
> 
> ##### Abnormal Security - Incoming Mapper
> 
> - Updated the Abnormal Security - Incoming Mapper mapper to include genai_summary field in case details.
> 

Reviewer: ```suggestion

#### Incident Fields

##### New: Abnormal Security Case GenAI Summary

- New: Added the *Abnormal Security Case GenAI Summary* incident field.

#### Integrations

##### Abnormal Security
- Updated the Docker image to: *demisto/python3:3.12.8.1983910*.

- Added support for the following parameters:
    - **Maximum incidents pages to fetch** that limits the number of incidents fetched in a single fetch_incidents call.
    - **Polling Lag Time (in minutes)** that adds a custom lag while polling incidents.
- Added support for the following arguments in **abnormal-security-get-threat** command, to enable pagination:
    - *page_number*
    - *page_size*
- Fixed pagination issue in fetch-incidents to handle max_fetch below 100 properly.
- Enhanced pagination support in fetch-incidents to reliably fetch more than 100 incidents across multiple pages, enabling comprehensive data retrieval for large-scale incident analysis.

  
#### Mappers

##### Abnormal Security - Incoming Mapper

- Updated the Abnormal Security - Incoming Mapper to include genai_summary field in case details.

```

Author: done
